pkg/database/sqlite_db_factory.go

```diff
@@ -0,0 +1,147 @@
+package database
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	glog "log"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/rotisserie/eris"
+	"golang.org/x/exp/slices"
+
+	lru "github.com/hashicorp/golang-lru/v2"
+	"github.com/mattn/go-sqlite3"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+	"gorm.io/plugin/dbresolver"
+)
+
+// sqliteDbFactory will make Sqlite DbInstance.
+type sqliteDbFactory struct {
```

Comment:
sorry, but this is not a factory, this is SqliteAdapter or SQliteInstance or just `Sqlite`.

Reply:
it is a private builder of the instance but is not itself the instance